lib/merch_calendar/fiscal_year_calendar.rb

```diff
@@ -2,117 +2,89 @@
+      end
     end
-
-    # Return the ending date for a particular quarter
+    
     def end_of_quarter(year, quarter)
-      @retail_calendar.end_of_quarter(*offset_quarter(year, quarter))
-    end
-
-    # The date of the first day of the merch month
-    # @param [Fixnum] year - the fiscal year
-    # @param [Fixnum] merch_month - the nth month of the offset calendar
-    #   ex) for an offset of +/- 2 quarters, month 1 = August
-    def start_of_month(year, merch_month)
-      @retail_calendar.start_of_month(*offset_month(year, merch_month))
-    end
-
-    # The date of the last day of the merch month
-    # @param [Fixnum] year - the fiscal year
-    # @param [Fixnum] merch_month - the nth month of the offset calendar
-    #   ex) for an offset of +/- 2 quarters, month 1 = August
-    def end_of_month(year, merch_month)
-      @retail_calendar.end_of_month(*offset_month(year, merch_month))
-    end
-
-    # Returns the date that corresponds to the first day in the merch week
-    def start_of_week(year, merch_month, merch_week)
-      @retail_calendar.start_of_week(*offset_month(year, merch_month), merch_week)
-    end
-
-    # Returns the date that corresponds to the last day in the merch week
-    def end_of_week(year, merch_month, merch_week)
-      @retail_calendar.end_of_week(*offset_month(year, merch_month), merch_week)
-    end
-
-    # Returns the number of weeks in the fiscal year
-    def weeks_in_year(year)
-      @retail_calendar.weeks_in_year(offset_year(year))
+      case quarter
+      when QUARTER_1
+        end_of_month(year, 3)
+      when QUARTER_2
+        end_of_month(year, 6)
+      when QUARTER_3
+        end_of_month(year, 9)
+      when QUARTER_4
+        end_of_month(year, 12)
+      end
     end
 
-    private
+    def end_of_year(year)
+      year_end = Date.new((year), 7, -1) # July 31st
+      wday = (year_end.wday + 1) % 7
 
-    # Offsets the quarter based on the fiscal year quarter offset
-    # returns: offset [year, quarter]
-    def offset_quarter(year, quarter)
-      # first quarter in fiscal calendar is Q3 of retail calendar of previous year
-      if quarter >= 1 + @fy_quarter_offset.abs 
-        [year, quarter + @fy_quarter_offset]
+      if wday > 3
+        year_end += 7 - wday
       else
-        [year - 1, quarter - @fy_quarter_offset]
+        year_end -= wday
       end
+      year_end
     end
-
-    # Offsets the month based on the fiscal year quarter offset
-    # returns: offset [year, month]
-    def offset_month(year, month)
-      # 3 - number of months in a quarter
-      month_offset = @fy_quarter_offset * 3
-
-      if month >= (month_offset.abs + 1) 
-        [year, month + month_offset] 
-      else
-        [year - 1, month - month_offset]
+    
+    def start_of_month(year, merch_month)
+      # 91 = number of days in a single 4-5-4 set 
+      start = start_of_year(year) + ((merch_month - 1) / 3).to_i * 91
+
+      case merch_month
+      when 2,5,8,11
```

Comment:
The code in the retail calendar should not be taken as a good example, as it uses a lot of "magic numbers".  Since you're rewriting this, anywhere you see a number, please extract it to a constant.